code/xcvm/cosmwasm/contracts/gateway/src/contract/execute.rs

```diff
@@ -71,8 +71,26 @@ fn handle_config_msg(auth: auth::Admin, deps: DepsMut, msg: ConfigSubMsg, env: E
 		ConfigSubMsg::ForceAssetToNetworkMap { this_asset, other_network, other_asset } =>
 			assets::force_asset_to_network_map(auth, deps, this_asset, other_network, other_asset),
 		ConfigSubMsg::ForceNetwork(msg) => network::force_network(auth, deps, msg),
-		ConfigSubMsg::ForceInstantiate { user_origin, salt } =>
-			interpreter::force_instantiate(auth, env.contract.address, deps, user_origin, salt),
+		ConfigSubMsg::ForceInstantiate { user_origin, salt } => interpreter::force_instantiate(
+			auth,
+			env.contract.address.clone(),
+			deps,
+			user_origin,
+			salt,
+		),
+		ConfigSubMsg::Force(msgs) => {
+			let mut aggregated = Response::new();
+			for msg in msgs {
+				let response = handle_config_msg(auth, deps.branch(), msg, env)?;
+				ensure!(
+					response.attributes.is_empty() && response.data.is_none(),
+					ContractError::BatchedCallsCannotReturnData
+				);
```

Comment:
The correct way to deal with this not being allowed is making it impossible to express in code for example by introducing:

```rust
struct ConfigMsgResponse {
    pub messages: Vec<SubMsg>,
    pub events: Vec<Event>,
}

impl ConfigMsgResponse {
    pub fn add_message(mut self, msg: impl Into<CosmosMsg>) -> Self {
        self.messages.push(SubMsg::new(msg))
        self
    }

    pub fn add_submessage(mut self, msg: SubMsg<T>) -> Self {
        self.messages.push(msg);
        self
    }

    pub fn add_event(mut self, event: Event) -> Self {
        self.events.push(event);
        self
    }

    pub fn merge(&mut self, other: Self) {
        self.messages.extend(other.messages.drain(..));
        self.events.extend(other.events.drain(..));
    }
}

impl From<ConfigMsgResponse> for Response {
    fn from(resp: ConfigMsgResponse) -> Self {
        let mut result = Self::new();
        result.messages = resp.messages;
        result.events = resp.events;
        result
    }
}
```rust

and then, after changing all methods to return `Result<ConfigMsgResponse>`, this method will be something along the lines of:

```rust
fn handle_config_msg(auth: auth::Admin, mut deps: DepsMut, msg: ConfigSubMsg, env: &Env) -> Result {
	match msg {
		ConfigSubMsg::ForceAsset(msg) => assets::force_asset(auth, deps, msg),
		...,
		ConfigSubMsg::Force(msgs) => {
			let mut aggregated = ConfigMsgResponse::new();
			for msg in msgs {
				aggregated.merge(handle_config_msg(auth, deps.branch(), msg, env)?);
			}
			OK(aggregated)
		},
	}.map(Response::from)
}
```

Reply:
i can do it in next prs if patter will repeat. for me current approach is minimally intrusive and aligns with cw coding practices. 

i agree  that special object is eventually better, but i am pretty sure this code is half complete for seamless replacement of buildin response with custom, and in the end can end up that why use response at all, we could use custom non reponse object to instance only topology, not instanti which has message. so in theory no need messages too. 

so i agree with approach in general, but i do not think it should be done now.

Reply:
The current approach is less code but it exposes and internal bug as an external contract error.  It introduces failure condition which can easily be avoided.  I don’t see why this shouldn’t be done now.  If messages aren’t needed either that’s even better since messages handling can be removed as well.

Reply:
message is needed for  creating contracts, interpreter. admin creates interpreter. in theory can refactor out that method. because i do not really need it in deploy, but that is other think. so to have batch need to change many things - a lot . to make it typesafe. which is not CW coding practice.

if CW will provide batch, it may be as I coded.

Reply:
btw,
```rust
struct BatchMsgResponse {
    inner: Response,
}
```
could be even better

Reply:
You don’t need to change a lot of things to have this be type-safe.  You need `ConfigMsgResponse` as in my previous comment and then change `force_` methods to return it rather than `Response`.  That’s not such a high price to pay when it eliminates an error-prone construct which this PR exposes as part of public API.